exonum/src/blockchain/service.rs

```diff
@@ -400,19 +402,39 @@ impl SharedNodeState {
 
         lock.peers_info.clear();
         lock.majority_count = state.majority_count();
+        lock.is_validator = state.is_validator();
+        lock.validators.clear();
 
-        for (p, c) in state.peers().iter() {
+        for (p, c) in state.peers() {
             lock.peers_info.insert(c.addr(), *p);
         }
+        for v in state.validators() {
+            lock.validators.push(*v);
+        }
     }
 
-    /// Returns the majority count from the current "State"
-    /// of a blockchain node.
-    pub fn majority_count(&self) -> usize {
-        self.state
-            .read()
-            .expect("Expected read lock.")
-            .majority_count
+    /// Returns a boolean value which indicates whether the consensus is
+    pub fn is_consensus(&self) -> bool {
+        let lock = self.state.read()
+            .expect("Expected read lock.");
+
+        let mut active_validators: usize = 0;
+        for peer_key in lock.peers_info.values() {
+            if lock.validators.iter()
+                .any(|validator| validator.consensus_key == *peer_key) {
+                active_validators += 1;
+            }
+        }
+
+        if lock.is_validator {
+            // Peers list doesn't include current node address, so we have to increment its length.
+            // E.g. if we have 3 items in peers list, it means that we have 4 nodes overall.
+            active_validators += 1;
+        }
+
+        // Just after Node is started (node status isn't updated) majority_count = 0,
+        // and as result is_consensus() will return true.
```

Comment:
I guess statement `and as result is_consensus() will return true` isn't quite correct.
I'd suggest `so we have to check that majority count is greater than 0.`.